File: sorting-algos/bubblesort/bubblesort.py
def swap(i, j, arr):
    temp = arr[i]
    arr[i] = arr[j]
    arr[j] = temp

def isSorted(arr):
    for i in range(len(arr)-1):
        if arr[i] > arr[i+1]:
            return False
    return True

def bubble(arr, end):
    index = 0 
    while index <= end:
        if arr[index] > arr[index+1]:
            swap(index, index+1, arr)
        index += 1 

def bubbleSort(arr):
    end = len(arr)-2
    while not isSorted(arr):
        bubble(arr, end)
        end -= 1

arr = [5,4,7,8,9,1,4,3,10,12]
bubbleSort(arr)
print(arr)

File: sorting-algos/bubblesort/bubblesort.cpp
#include <iostream>
using namespace std;

void printArr(int arr[], int size){
    int i=0;
    while(i < size){
        cout << arr[i++] << " ";
    }
    cout << "\n";
}

void swap(int i, int j, int* arr){
    int TEMP = arr[i];
    arr[i] = arr[j];
    arr[j] = TEMP; 
}

bool isSorted(int arr[], int size){
    int index=0;
    while(index < size-1){
        if(arr[index] > arr[index+1]){
            return false;
        }
        index++;
    }
    return true;
}

void bubble(int* arr, int end){
    int index = 0;
    while(index < end-1){
        if(arr[index] > arr[index+1]){
            swap(index, index+1, arr);
        }
        index++;
    }
}

void bubblesort(int* arr, int size){
    int end = size;
    while(!isSorted(arr, size)){
        bubble(arr, end--);
    }
}

int main(){
    int arr[] = {4,6,2,8,9,86,12,34,1,0};
    int arraySize = sizeof(arr)/sizeof(arr[0]);
    bubblesort(arr, arraySize);
    int i=0;
    while(i < arraySize){
        cout << arr[i++] << " ";
    }
    return 0;
}